src/ng/compile.js

```diff
@@ -3350,9 +3459,56 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
       }
     }
 
+    function getTrustedPropContext(nodeName, propNormalizedName) {
+      var prop = propNormalizedName.toLowerCase();
+      return PROP_CONTEXTS[nodeName + "|" + prop] || PROP_CONTEXTS["*|" + prop];
+    }
+
+    function sanitizeSrcsetPropertyValue(value) {
+      return sanitizeSrcset($sce.valueOf(value), 'ng-prop-srcset');
+    }
+    function addPropertyDirective(node, directives, attrName, propName) {
+      if (EVENT_HANDLER_ATTR_REGEXP.test(propName)) {
+        throw $compileMinErr('nodomevents',
+            'Property bindings for HTML DOM event properties are disallowed.  Please use the ' +
+                'ng- versions (such as ng-click or ng-on-click instead of ng-prop-onclick) instead.');
+      }
+
+      var nodeName = nodeName_(node);
+      var trustedContext = getTrustedPropContext(nodeName, propName);
+
+      var sanitizer = identity;
+      // Sanitize img[srcset] + source[srcset] values.
+      if ((nodeName === 'img' || nodeName === 'source') && propName === 'srcset') {
+        sanitizer = sanitizeSrcsetPropertyValue;
+      } else if (trustedContext) {
+        sanitizer = $sce.getTrusted.bind($sce, trustedContext);
+      }
+
+      directives.push({
+        priority: 100,
+        compile: function ngPropCompileFn(_, attr) {
+          var fn = $parse(attr[attrName]);
+          return {
+            pre: function ngPropPreLinkFn(scope, $element) {
+              scope.$watch(fn, function propertyWatchActionFn(value) {
```

Comment:
Also, although the SCE stuff should take care of sanitising and protecting things like `innerHTML` I think it would be sensible to have a few tests here either to check that the call to `$sce,getTrusted` is made or that `innerHTML` is sanitised correctly.

Reply:
I still need to write a lot of sce related tests

Reply:
I think https://github.com/angular/angular.js/pull/16614/commits/f547b0da82724b1d9d2f3ec38f8c7332bd4f1933 covers the specific cases you mentioned?